packages/app-mobile/root.tsx

```diff
@@ -180,6 +182,12 @@ const generalMiddleware = (store: any) => (next: any) => async (action: any) =>
 		await AlarmService.updateNoteNotification(action.id, action.type === 'NOTE_DELETE');
 	}
 
+	if (action.type === 'NOTE_DELETE' && newState.route?.routeName === 'Note' && newState.route.noteId === action.id) {
+		const parentItem = action.originalItem?.parent_id ? await Folder.load(action.originalItem?.parent_id) : null;
+		const parentId = getDisplayParentId(action.originalItem, parentItem);
+		await NavService.go('Notes', { folderId: parentId });
+	}
```

Comment:
This pull request is currently a draft, until this is replaced with a better way to handle post-deletion navigation.

At present, the `deleteNote` and `permanentlyDeleteNote` commands assume that note screen navigation is handled by other parts of the app. For example, on desktop, deleting the focused note removes it from the note list, which moves the note viewer to a different note.

On mobile, however, the note deletion logic previously **relied a `NAV_GO` to show the note screen** after deleting the current note. This pull request moves the `NAV_GO` to the reducer.

I dislike the `routeName === 'Note' && newState.route.noteId === action.id` comparison — the reducer now checks state information specific to `Note.tsx`, which could make the note screen more difficult to maintain and test.